WizardSchoolSignUpUIKit/WizardSchoolSignUpUIKit/ViewController.swift

```diff
@@ -8,12 +8,63 @@
 import UIKit
 
 class ViewController: UIViewController {
-
+    @IBOutlet private var userNameTextField: UITextField!
+    @IBOutlet private var userNameIcon: UIImageView!
+    @IBOutlet private var passwordTextField: UITextField!
+    @IBOutlet private var passwordIcon: UIImageView!
+    @IBOutlet private var passwordConfirmationTextField: UITextField!
+    @IBOutlet private var passwordConfirmationIcon: UIImageView!
+    @IBOutlet private var createAccountButton: UIButton!
+    
+    private var userName = ""
+    private var password = ""
+    
+    private var network = Networking()
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         // Do any additional setup after loading the view.
-    }
-
+        userNameTextField.delegate = self
+        passwordTextField.delegate = self
+        passwordConfirmationTextField.delegate = self
 
+    }
+    
+    
 }
 
+// MARK: - Text Field Delegate
+extension ViewController: UITextFieldDelegate {
+    func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
+        
+        if textField == userNameTextField, let userName = userNameTextField.text {
+            
+            network.validate(userName: userName) { (result) in
+                switch result {
+                
+                case .success(let isAvailable):
+                    DispatchQueue.main.async { [weak self] in
+                        if isAvailable {
+                            self?.userNameIcon.tintColor = UIColor.green
+                        } else {
+                            self?.userNameIcon.tintColor = UIColor.red
+                        }
+                    }
+                case .failure(_):
+                    break
+                }
+            }
```

Comment:
I'd suggest for us to extract this into a method like:
```suggestion
            validate(userName: userName)
```

```swift
    private func validate(userName: String) {
        network.validate(userName: userName) { result in
            switch result {
            case .success(let isAvailable):
                DispatchQueue.main.async { [weak self] in
                    self?.userNameIcon?.tintColor = isAvailable ? .systemGreen : .systemRed
                }
            case .failure:
                break
            }
        }
    }
```

Also updated the `tintColor` to use semantic colors so we can get system default colors for Dark Mode.